contracts/uniswap_exchange.vy

```diff
@@ -63,32 +79,31 @@ def addLiquidity(min_liquidity: uint256, max_tokens: uint256, deadline: timestam
     total_liquidity: uint256 = self.totalSupply
     if total_liquidity > 0:
         assert min_liquidity > 0
-        eth_reserve: uint256(wei) = self.balance - msg.value
+        eth_reserve: uint256 = as_unitless_number(self.balance - msg.value)
         token_reserve: uint256 = self.token.balanceOf(self)
-        token_amount: uint256 = msg.value * token_reserve / eth_reserve + 1
-        liquidity_minted: uint256 = msg.value * total_liquidity / eth_reserve
+        # Platform profit is in ppm so we divide the liquidity minted by a 1000
+        platform_profit: uint256 = self.sqrt(1000000*eth_reserve*token_reserve/self.previous_invariant)
```

Comment:
First we calculate the profit of the platform in ppm by taking the square root of the growth of the invariant

Reply:
And multiplying it by 1M